File: solutions/implementation/bunga-gabungan/bunga-gabungan.cpp
/**
 * Solution by Reydeus
 * Title: Bunga Gabungan
 * Level: 300
 * Original Problem: https://tlx.toki.id/problems/troc-19/A
 * Submission: https://tlx.toki.id/problems/troc-19/A/submissions/2419846
 */

#include <iostream>

int main ()
{
  int p, q;
  std::cin >> p, q;

  int sum = p*p + q*q + 1;

  if (sum % 4 == 0)
  {
    std::cout << sum / 4 << std::endl;
    return 0;
  }
  else {
    std::cout << "-1" << std::endl;
    return 0;
  }
}

File: tlx/data-structure/implementation/bunga-gabungan/bunga-gabungan.py
# Submission & solution by Reydeus
# Title: Bunga Gabungan
# Level: 300
# Original Problem: https://tlx.toki.id/problems/troc-19/A
# Submission: https://tlx.toki.id/submissions/2404993

a, b = map(int, input().split())
sum = a*a + b*b + 1

print(f"{sum / 4}" if sum % 4 == 0 else "-1")
